scripts/globals.py

```diff
@@ -2,26 +2,54 @@
 import os
 
 import modules.scripts as scripts
-import modules.shared as shared
+from modules.shared import opts, models_path, cmd_opts
+from modules.cmd_args import parser
 
 
-global hide_nsfw_networks_key, hide_nsfw, network_descriptions_path
+global hide_nsfw
 
 
-hide_nsfw_networks_key = "sd_lora_tagger_hide_nsfw_extra_networks"
-network_descriptions_path = os.path.join(scripts.basedir(), "network_descriptions")
+def out(msg):
+    print(f"SD Lora Tagger: {msg}")
+
+
+""" Paths """
+lora_tagger_dir = scripts.basedir()
+models_dir = models_path
+network_descriptions_path = os.path.join(lora_tagger_dir, "network_descriptions")
 
-print(f"SD Lora Tagger: network_descriptions_path={network_descriptions_path}")
+# Used in accordance to https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/cmd_args.py
+model_dirs = {
+    "Lora": cmd_opts.lora_dir,
+    "LyCORIS": cmd_opts.lyco_dir,
+    "embeddings": cmd_opts.embeddings_dir,
+    "hypernetworks": cmd_opts.hypernetwork_dir,
+    "Stable-diffusion": cmd_opts.ckpt_dir
```

Comment:
`cmd_opts.lyco_dir` errors if the extension is not installed. It should probably be added on run-time when checking if the extension is installed.

Reply:
Fixed the error by checking ```if hasattr(cmd_opts, "lyco_dir")```